frontend/src/modules/Dashboard/Components/Dashboard.tsx

```diff
@@ -28,17 +28,34 @@ type FilterOption =
   | 'matchable'
   | 'no-check-in-email'
   | 'no-no-match-email'
+const filterOptionDisplay = [
+  ['no-filter', 'All Classes'],
+  ['unmatchable', 'Unmatchable'],
+  ['newly-matchable', 'Newly Matchable'],
+  ['matchable', 'Matchable'],
+  ['no-check-in-email', 'No Check In Email'],
+  ['no-no-match-email', 'No No Match Email'],
+]
+const sortOrderDisplay = [
+  ['newest-requests-first', 'Newest Requests First'],
+  ['classes-a-z', 'Classes A-Z'],
+  ['classes-z-a', 'Classes Z-A'],
+]
 
 export const Dashboard = () => {
+  const history = useHistory()
   const { user } = useAuthValue()
   const { courses } = useCourseValue()
   const { students } = useStudentValue()
-
-  const [filteredOption, setFilteredOption] = useState<FilterOption>(
-    'no-filter'
+  const state = history.location.state as {
+    sortedOrder: SortOrder
+    filterOption: FilterOption
+  }
+  const [sortedOrder, setSortedOrder] = useState<SortOrder>(() =>
+    state?.sortedOrder ? state.sortedOrder : 'newest-requests-first'
   )
-  const [sortedOrder, setSortedOrder] = useState<SortOrder>(
-    'newest-requests-first'
+  const [filteredOption, setFilteredOption] = useState<FilterOption>(() =>
+    state?.filterOption ? state.filterOption : 'no-filter'
   )
```

Comment:
Super specific nit, but for better readability/easier changes, you can make constant variables for "default sorting order" and "default no filter" in place of the "newest-requests-firsts" and "no-filter"